### DISCUSSION TITLE
Unresolved dependency issue while upgrading to 2.9.0-RC2 from 2.8

### DISCUSSION THREAD

**Discussion by Harish247** [playframework/playframework#11983] [Q&A]
I am trying 2.9.0 ahead with RC2. During the project built I have seen some libraries referencing with an extra **_3**. I am not sure what it is referencing to. For example

```
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: com.iterable#swagger-play_3;2.0.1: not found
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
```

The original dependency I have is **com.iterable#swagger-play;2.0.1**
But for some reason play 2.9.0 throwing un resolved dependency by adding **_3** at the end. 
I had another selenium remote driver dependency with the same issue.

SBT version: 1.9.6
Scala version: 3.3.1, but the issue persists with 2.8.1 also
Java version: 11.0.1
Play version: 2.9.0-RC2

Please suggest any solutions. Thanks

**Comment by mkurz** [comment]
This "UNRESOLVED DEPENDENCIES" error occurs because you are using Scala 3, but the libraries you want to use are not published for Scala 3 yet. Look here at the bottom: https://repo1.maven.org/maven2/com/iterable/ -> swagger-play is "only" published for Scala 2.13, 2.12, etc.
According to https://github.com/Iterable/swagger-play this module is deprecated anyway in favor or https://github.com/dwickern/swagger-play
You may want to switch to that repo instead and reach out to the maintainer to publish the library for Scala 3.

BTW: Using Scala 3 with Play 2.8 will _never_ work because we do not support Scala 3 in Play 2.8. To use Scala 3, Play 2.9 is _required_.

Possible solutions for you are now:
* Keep using Scala 2.13.12
* If you want to use Scala 3 with libraries that are not published for Scala 3 yet you should be able to do so by using the cross-version workaround, see https://www.scala-sbt.org/1.x/docs/Cross-Build.html#Scala+3+specific+cross-versions:
```sbt
 libraryDependencies +=
      ("com.iterable" %% "swagger-play" % "2.0.1").cross(CrossVersion.for3Use2_13)
```

Hope that helps!

**Comment by Harish247** [comment]
Thanks for the suggestion. It works if I use Scala version 2.13.9.
I am using play 2.9.0-RC2.
